src/main.c

```diff
@@ -173,7 +173,7 @@ int main(int argc, char **argv) {
     fprintf(stderr, "Error creating lite-xl window: %s", SDL_GetError());
     exit(1);
   }
-  ren_init(window);
+  window_renderer = ren_init(window);
```

Comment:
What do you think about making `ren_init` handle `window_renderer` internally?
So that `main.c` doesn't need to care about it, other than just calling `ren_free`.
Maybe even moving the "main" window creation code inside `ren_init`.

I think this would also make sense with multi-window in mind, so that `main.c` doesn't need to care about, and manage, eventual multiple windows.

Reply:
> What do you think about making `ren_init` handle `window_renderer` internally?

ren_init creates a renderer given an SDL Window.
If it handled the resulting renderer on its own, we would have trouble managing multiple windows, which I intend to do entirely in Lua.

> So that `main.c` doesn't need to care about it, other than just calling `ren_free`. Maybe even moving the "main" window creation code inside `ren_init`.

This is a stop gap until Windows are properly implemented as Lua metatables.

> I think this would also make sense with multi-window in mind, so that `main.c` doesn't need to care about, and manage, eventual multiple windows.

Agreed, but one step at a time.
this change is very small and would need to be included for multi-window anyways since its saner to simply push pointers to the stack as userdata and apply a metatable onto it, like dirmonitor does.

Reply:
I mean to just avoid making `main.c` aware of `window_renderer`, so that you just need to call `ren_init` without getting a returned value, and `ren_free` without passing it anything.
And ideally also moving the current window creation code inside `ren_init`.

Reply:
> I mean to just avoid making `main.c` aware of `window_renderer`, so that you just need to call `ren_init` without getting a returned value, and `ren_free` without passing it anything. And ideally also moving the current window creation code inside `ren_init`.

`ren_init` needs to be converted to return pointers anyways, so I don't see the issue.